### PR TITLE
[RDY] fix bc/gst tester for istanbul

### PR DESCRIPTION
**Pull request by jangko**
waiting for #417 and #418 merged

### CODE REVIEW

**Comment by jangko** [comment]
this PR is ready for review.

File: tests/test_generalstate_json.nim [hunk 1]
```diff
@@ -175,15 +179,17 @@ proc generalStateJsonMain*(debugMode = false) =
   if paramCount() == 0 or not debugMode:
     # run all test fixtures
     suite "generalstate json tests":
-      jsonTest("GeneralStateTests", testFixture)
+     jsonTest("GeneralStateTests", testFixture)
```
Comment (stefantalpalaru) [inline tests/test_generalstate_json.nim]: Wrong indentation here?

File: tests/test_blockchain_json.nim [hunk 2]
```diff
@@ -663,14 +669,16 @@ proc main() =
...
+      jsonTest("newBlockchainTests", testFixture)
   else:
     # execute single test in debug mode
     let config = getConfiguration()
     if config.testSubject.len == 0:
       echo "missing test subject"
       quit(QuitFailure)
 
-    let path = "tests" / "fixtures" / "BlockChainTests"
+    let path = "tests" / "fixtures" / "newBlockChainTests"
```
Comment (stefantalpalaru) [inline tests/test_blockchain_json.nim]: This means that when in debug mode you only run half the tests? I saw the same being done in "tests/test_generalstate_json.nim".

File: tests/test_generalstate_json.nim (hunk 1 above)
Comment (jangko) [reply tests/test_generalstate_json.nim]: fixed

File: tests/test_blockchain_json.nim (hunk 2 above)
Comment (jangko) [reply tests/test_blockchain_json.nim]: in debug mode, only one json file is executed, one EVM fork selected and one specific subtest selected. perhaps we can use cli to configure the input path or reorganize test fixtures folders.

File: tests/test_generalstate_json.nim (hunk 1 above)
Comment (stefantalpalaru) [reply tests/test_generalstate_json.nim]: But why didn't it crash at compilation time?

Comment (jangko) [reply tests/test_generalstate_json.nim]: a block with single line body is allowed to have any indentation GTE one. or in other words the first line of the block body will set the valid indentation for the whole body.

**Pull request merged**